src/acrn/acrn_driver.c

```diff
@@ -2562,11 +2562,21 @@ acrnStateInitialize(bool privileged,
                     virStateInhibitCallback callback ATTRIBUTE_UNUSED,
                     void *opaque ATTRIBUTE_UNUSED)
 {
+    int fd;
+
     if (!privileged) {
         VIR_INFO("Not running privileged, disabling driver");
         return 0;
     }
 
+    /* check if we are running on an ACRN enabled system */
+    fd = acrnGetVhmFd();
+    if (fd < 0) {
+        VIR_INFO("ACRN hypervisor not available, disabling driver");
+        return 0;
+    }
+    close(fd);
+
```

Comment:
Since acrnInitPlatform() is called after this, how about if we do the following?

- make acrnGetPlatform() return error codes (and perhaps -ENODEV in this case)
- make acrnInitPlatform() return error codes
- print a message and return 0 when acrnInitPlatform() indicates this is not an ACRN system